src/content/reference/react-dom/hooks/useFormState.md

```diff
@@ -0,0 +1,219 @@
+---
+title: useFormState
+canary: true
+---
+
+<Canary>
+
+The `useFormState` Hook is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`useFormState` is a Hook that allows you to show error messages or other responses after submitting a form.
+
+```js
+const [state, formAction] = useFormState(action, initalState);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `useFormState()` {/*useformstate*/}
+
+Call `useFormState` at the top level of your component to see the return value of an action after submitting a form. You pass `useFormState` an existing action as well as an initial state, and it returns a new action that you use when submitting your form, along with the latest form state.
+
+```js
+function AddToCart({itemID}) {
+  const [message, formAction] = useFormState(addToCartAction, null);
+  return (
+    <form action={formAction}>
+      <input type="hidden" name="itemID" value={itemID} />
+      <button type="submit" label="Add to cart" />
+      <p>
+        {message}
+      </p>
+    </form>
+  )
+}
+```
+
+The form state is the value returned by the action when the form was last submitted. If the form has not yet been submitted, it is the initial state that you pass.
+
+If used with a server action, `useFormState` allows the server's response from submitting the form to be shown even before hydration has completed.
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `action`: The action to be performed when the form is submitted. It can be a server or a client action. When the action is called, it will receive the previous state of the form (initially the `initialState` that you pass, subsequently its previous return value) as its initial argument, followed by the arguments that an action normally receives.
+* `initialState`: The value you want the state to be initially. It can be any serializable value. This argument is ignored after the form is first submitted.
+
+
+#### Returns {/*returns*/}
+
+`useFormState` returns an array with exactly two values:
+
+1. The current state. During the first render, it will match the `initialState` you have passed. After the form is submitted, it will match the value returned by the action.
+2. A new action that you can pass as the `action` prop to your `form` component.
+
+---
+
+## Usage {/*usage*/}
+
+### Display a message after submitting a form {/*display-a-message-after-submitting-a-form*/}
```

Comment:
Can you reword the title to something more use-case based? I think that could help folks that are trying to accomplish a certain task find this e.g. Displaying form errors